src/components/CardGridSearch.astro

```diff
@@ -0,0 +1,255 @@
+				const headings = Array.from(this.children).filter(
+					(el) => el instanceof HTMLHeadingElement || el instanceof HTMLParagraphElement
+				);
+				// Create "nothing found" for each heading
+				headings.forEach((heading, i) => {
+					const wrn = document.createElement('p');
+					wrn.classList.add('hidden', 'warning');
+					wrn.id = `warn-${i}`;
+					wrn.innerHTML = `
+						<style>
+							.warning {
+								color: var(--sl-color-orange-high);
+								font-size: larger;
+							}
+							.hidden {
+								display: none;
+							}
+						</style>
+						<span>Nothing found in ${heading.textContent}</span>
+						`;
+					heading.insertAdjacentElement('afterend', wrn);
+				});
+				// Select all "nothing found" created before
+				const warnings = document.querySelectorAll('[id^="warn-"]');
+
+				// Prevent page reload
+				form.addEventListener('submit', (e) => {
+					e.preventDefault();
+				});
+
+				// Prevent reload on 'Enter', and enable clean input on 'Escape'
+				form.addEventListener('keydown', (e) => {
+					clear.classList.remove('pagefind-ui__suppressed');
+					if (e.key === 'Escape') {
+						reset();
+					} else if (e.key === 'Enter') e.preventDefault();
+				});
+				searchBar.addEventListener('keydown', (e) => {
+					if (e.key === 'Enter') e.preventDefault();
+				});
+
+				// Search
+				searchBar.addEventListener('keyup', () => {
+					searchFilter();
+				});
+				// Clear search button 'X'
+				clear.addEventListener('click', () => {
+					clear.classList.add('pagefind-ui__suppressed');
+					reset();
+				});
+
+				function reset() {
+					searchBar.value = '';
+					searchFilter();
+				}
+
+				/**
+				 * Function to search and filter based on input
+				 */
+				function searchFilter() {
+					const query = searchBar.value.toUpperCase();
+
+					inputs.forEach((elements, i) => {
+						const cards = Array.from(elements.children);
+						let total = cards.length;
+
+						cards.forEach((el) => {
+							const htmlElement = el as HTMLElement;
+							const span = el.querySelector('span');
+							const txtValue = span?.textContent || span?.innerText;
+							if (!txtValue) return;
+
+							// Filter list
+							if (txtValue.toUpperCase().indexOf(query) > -1) {
+								total++;
+								htmlElement.style.display = '';
+							} else {
+								total--;
+								htmlElement.style.display = 'none';
+							}
+
+							// Show warning if none found
+							if (total === 0) {
+								warnings[i].classList.remove('hidden');
+							} else {
+								warnings[i].classList.add('hidden');
+							}
+						});
+					});
+				}
+			});
+		}
+	}
+	customElements.define('card-search', CardSearch);
+</script>
+
+<style>
+	:root {
+		/* Starlight */
+		--pagefind-ui-primary: var(--sl-color-accent-light);
+		--pagefind-ui-text: var(--sl-color-gray-2);
+		--pagefind-ui-font: var(--__sl-font);
+		--pagefind-ui-background: var(--sl-color-black);
+		--pagefind-ui-border: var(--sl-color-gray-5);
```

Comment:
Since the logic for this component is 100% created by us then I think it makes sense to instead use the respective `--sl-*` vars directly instead of re-defining them with `--pagefind-ui-*` and then using that below again.

Reply:
https://github.com/tauri-apps/tauri-docs/pull/1740/commits/67d5bdca3b8efe355ad2053a76b63f271a8f0e0d